haystack/nodes/preprocessor/preprocessor.py

```diff
@@ -444,3 +457,49 @@ def _find_longest_common_ngram(
             # no common sequence found
             longest = ""
         return longest if longest.strip() else None
+
+    def _sentence_tokenizer(
+        self, text: str, language: str, tokenizer_model_folder: Optional[Union[str, Path]] = None
+    ) -> List[str]:
+        """
+        Tokenize text into sentences.
+        :param text: str, text to tokenize
+        :param language: str, language of text
+        :param tokenizer_model_path: Path, path to tokenizer model
+        :return: list[str], list of sentences
+        """
+        sentences = []
+        language_name = iso639_to_nltk.get(language)
+        tokenizer_model_path = (
+            Path(tokenizer_model_folder) / f"{language}.pickle" if tokenizer_model_folder is not None else None
```

Comment:
What if `language` is `None` here? Do you think we can give a default, or is better to check if it's `None` before building this path?

In general I think the whole `if-else` chain can be simplified a bit if we check `if language:` early on.

Reply:
If `language` is `None`, `.get()` on the dict will return `None`.

Oh, I didn't set a default value at that point to avoid the usage of another variable, so we could log the different “behaviors” of the fallback:

1) Use default model for language X

2) Use custom model for language X
2.1) There is a default model for this language X in case of custom model lookup/format error.
2.2) There is no default model for this language X in case of error, and tokenizer fallback to English

Currently, we're logging all the possible scenarios with different messages.

```python
if not sentences and not language_name:
                logger.error(
                    f"PreProcessor couldn't find default or custom sentence tokenizer model for {self.language}. Using English instead."
                )
                sentences = nltk.tokenize.sent_tokenize(text, language="english")
elif not sentences and language_name:
                logger.error(
                    f"PreProcessor couldn't find custom sentence tokenizer model for {self.language}. Using default {self.language} model."
                )
                sentences = nltk.tokenize.sent_tokenize(text, language=language_name)
```

This part and the below it could be simplified with usage of `flags`, but the `linter` doesn't like it. If we want to keep the different error logs, we will need in flags and more ifs again. What do you think ? The more verbose error log could make easier the life of users (usually there is some sort of despair when they find errors).